grapher/barCharts/DiscreteBarChart.tsx

```diff
@@ -459,19 +475,21 @@ export class DiscreteBarChart
             : ""
     }
 
-    formatValue(series: DiscreteBarSeries): string {
+    formatValue(series: DiscreteBarSeries): Label {
         const column = this.yColumns[0] // todo: do we need to use the right column here?
         const { transformedTable } = this
 
         const showYearLabels =
             this.manager.showYearLabels || series.time !== this.targetTime
         const displayValue = column.formatValueShort(series.value)
-        return (
-            displayValue +
-            (showYearLabels
-                ? ` (${transformedTable.timeColumnFormatFunction(series.time)})`
-                : "")
-        )
+        return {
+            valueString: displayValue,
+            yearString: showYearLabels
+                ? ` in ${transformedTable.timeColumnFormatFunction(
```

Comment:
The proposition `in` works for years, but not for dates, e.g. `in April 28` is somewhat incorrect, it needs to be `on April 28`.

I think you can copy the implementation from the map tooltip here: https://github.com/owid/owid-grapher/blob/e925bd457c20da71aaf0d8ac5a279a63a430edee/grapher/mapCharts/MapTooltip.tsx#L188


I think something along these lines should work:

```suggestion
                ? ` ${transformedTable.getPreposition(transformedTable.timeColumn)} ${transformedTable.timeColumnFormatFunction(
```